BattleSim/src/main/java/com/revature/controllers/AccountController.java

```diff
@@ -91,6 +91,12 @@ public void login(HttpServletRequest request, HttpServletResponse response) thro
 			response.setStatus(401);
 		}
 	}
+	
+	public void logout(HttpServletRequest request, HttpServletResponse response) throws IOException {
+		HttpSession sess = request.getSession();
```

Comment:
Hey Drew, When you do request.getSession(); it actually generates a new session. You just need to add false as a parameter. so request.getSession(false);. Then all you need to do after is sess,invalidate(); and you can log out.

Reply:
Actually wait, I don't think you need false haha. I think it's implied.